app/pulp/app/serializers/fields.py

```diff
@@ -10,3 +11,36 @@ class RepositoryRelatedField(serializers.HyperlinkedRelatedField):
     view_name = 'repositories-detail'
     lookup_field = 'name'
     queryset = models.Repository.objects.all()
+
+
+class RepositoryNestedIdentityField(serializers.HyperlinkedIdentityField):
+    """
+    Used to represent an href identity field includes the associated repository name in the url.
```

Comment:
This was a change introduced in #2800 so when that merged, this commit is no longer part of this PR, I'll clean that up in another PR.